infrastructure/src/main/java/org/corfudb/infrastructure/LogUnitServer.java

```diff
@@ -298,10 +305,25 @@ private void handleException(Throwable ex, ChannelHandlerContext ctx, RequestMsg
      */
     @RequestHandler(type = PayloadCase.WRITE_LOG_REQUEST)
     private void handleWrite(RequestMsg req, ChannelHandlerContext ctx, IServerRouter router) {
+        final int uuidTailLength = 12;
+
         LogData logData = getLogData(req.getPayload().getWriteLogRequest().getLogData());
+        Map<UUID, Long> backPointerMap = logData.getBackpointerMap();
+
+        Map<String, Long> reducedBackPointerMap = new HashMap<>(backPointerMap.size());
```

Comment:
The memory related cost (allocation and GC) for this new HashMap is small since the map is small, but considering this is a very frequent operation, I wonder if we can do any better? such as a list of map entries in string format as `key + "=" + value`

Reply:
I'm not sure if we can/should improve anything here, the map is small and there is very little to optimize